parser/ndt.go

```diff
@@ -38,20 +36,38 @@ func (n *NDTParser) ParseAndInsert(meta map[string]bigquery.Value, testName stri
 		// Ignoring non-snaplog file.
 		return nil
 	}
+
+	if len(rawSnapLog) > 10*1024*1024 {
+		metrics.TestCount.With(prometheus.Labels{
+			"table": n.TableName(), "type": "oversize"}).Inc()
+		log.Printf("Ignoring oversize snaplog: %d, %s\n",
+			len(rawSnapLog), testName)
+		return nil
+	}
+
+	// Record the file size.
+	metrics.FileSizeHistogram.Observe(float64(len(rawSnapLog)))
```

Comment:
Suggest moving this above the filter so we can see the actual distribution of files before rejecting some?